content/rfcs/0001_distributed_execution_of_pipelines.md

```diff
@@ -84,85 +83,128 @@ The workers will need to be tagged with what kind of resource they are providing
 | Worker 2 | Windows 10 64bit |
 | Worker 3 | Debian Linux 64bit |
 
-When a pipeline is first created in needs to set on the pipeline creation window what
-kind of resources it requires. These tags will need to be made accessible by a drop down
-list for ease of usage. These tags can be created when a Worker is created and saved to Gaia.
-Tagging them can be done manually on the Worker Manager screen.
+Workers automatically detect on startup which pipeline languages (e.g. Go, Java, Python etc.) are supported
+and update their tags accordingly. Negative tags can be used to manually declare a specific language is not 
+support on this worker. During pipeline creation, custom tags can be set per pipeline. Pipelines with custom
+tags will be only executed on workers with at least the same set of tags.
 
 ## The Worker RPC API
 
-The Workers will talk to the Gaia Master via a set of defined RPC interfaces.
+The Workers will talk to the Gaia Primary instance via a set of defined gRPC interfaces.
 These are as follows:
 
 ```go
-// RegisterWorker will take a worker struct which contains the following information:
-// Security: This will be protected by the TLS connection between master and worker.
-// IP: The address of the worker
-// Name: The name of the worker which typically can be `hostname`.
-// Operating System: The OS of the worker to save as a label.
-// SDK: The SDK the worker has.
-rpc RegisterWorker(Worker) {}
-
-// RunPipeline will take a pipeline, and execute it. This ia bi-directional endpoint.
-// Pipeline struct:
-// ID: Id of the pipeline
-// Repo: The git repository for the pipeline. This is needed because the worker needs
-// to build the pipeline.
-rpc RunPipeline(Pipeline) returns (Success) {}
-
-rpc GetAllPipelines(Worker) returns (Pipelines) {}
+// WorkerInstance represents the identity of
+// a worker instance.
+message WorkerInstance {
+    string   unique_id    = 1;
+    int32    worker_slots = 2;
+    repeated string tags  = 3;
+}
+
+// PipelineRun represents one pipeline run.
+message PipelineRun {
+    string   unique_id     = 1;
+    int64    id            = 2;
+    string   status        = 3;
+    int64    start_date    = 4;
+	int64    finish_date   = 5;
```

Comment:
There are some very weird indentations in this. :D Can you fix that ? After that it's cool. :)